simview/broker.py

```diff
@@ -42,19 +52,16 @@ def broker(sim_type_list=[]):
         'box': '1'
     }
 
-    sim_ports = {
-        'A': '8080',
-        'B': '5551',
-        'C': '5552',
-        'D': '5553'
-    }
+    # generate dict of ports keyed by type
+    sim_ports = dict(zip(sim_type_list, sim_ports_list))
 
     poller = zmq.Poller()
 
     instructor_poller =  zmq.Poller()
     instructor_poller.register(instructor_pipe, zmq.POLLIN)
 
-    print('reg poller')
+    if debug:
+        print('registering poller')
 
     for i in range(0, len(sim_type_list)):
         print('starting simulation channel ' + str(i) + ' of type ' + str(sim_type_list[i]))
```

Comment:
line 69 `sim_ip = base_ip_address + simports[sim_type_list[i]]` 
you need to convert simports to a `str` here.